xdggs/h3.py

```diff
@@ -32,17 +58,22 @@ def from_variables(
     ) -> "H3Index":
         _, var, dim = _extract_cell_id_variable(variables)
 
-        resolution = var.attrs.get("resolution", options.get("resolution"))
-        return cls(var.data, dim, resolution)
+        grid_info = H3Info.from_dict(var.attrs | options)
+
+        return cls(var.data, dim, grid_info)
+
+    @property
+    def grid(self) -> H3Info:
```

Comment:
Replace `grid` by `grid_info`?

Shall we also add an "abstract" `grid` or `grid_info` property in the `DGGSIndex` base class?